### PR TITLE
Tweak default SDFGI settings for better quality

### PR DESCRIPTION
**Pull request by Calinou**
- Enable Read Sky Light to get proper outdoors lighting out of the box.
- Set bounce feedback to 0.5 by default to get a better quality result.
  - [Higher values may cause infinite feedback with bright surfaces.](https://github.com/godotengine/godot/issues/50773#issuecomment-886047017)
- Increase the number of frames to converge to improve quality at the cost of latency. Most scenes are fairly static after all.
- Use 75% Y scale by default as most scenes are not highly vertical.
- Reorder the Y scale enum to go from the lowest Y scale to the highest. Also rename the "Disabled" setting to "100%" for clarity.

**Edit:** This PR was amended in May 2021 to not tweak any performance-related settings. Therefore, performance should be roughly identical to what it was before. Performance tweaks were moved to https://github.com/godotengine/godot/pull/39965.

### CODE REVIEW

File: scene/resources/environment.cpp
Comment (aaronfranke) [inline scene/resources/environment.cpp:1321]: Why default to 75%?

Comment (Calinou) [reply scene/resources/environment.cpp:1321]: I discussed this with @reduz. Most level layouts aren't that vertical, and therefore benefit from having a slightly "flattened" probe structure. This should make leaks less noticeable.

We could go as far as defaulting to 50%, but 75% is probably a safer compromise.

**Comment by akien-mga** [comment]
Needs rebase after #40016.

**Review by Logharaa — APPROVED** [review]

**Review by clayjohn — APPROVED** [review]
This will still need reduz' final approval, but it looks fine.

**Comment by RaTcHeT302** [comment]
i don't think this commit will help at all, there's an issue with the algorithm itself at it's core

**Comment by Calinou** [comment]
@RaTcHeT302 The new settings definitely help in non-pathological situations. It's just that you apparently ran into a bug with SDFGI.

**Comment by aaronfranke** [comment]
@Calinou What's the status of this? It seems that this is desired, but this needs a rebase before it can be reviewed and merged.

**Comment by Calinou** [comment]
@aaronfranke reduz wasn't convinced by all of these default setting changes. I need to discuss this with him again first.

**Comment by reduz** [comment]
SDFGI needs further work on these areas, will work on it after Alpha

**Comment by Calinou** [comment]
> SDFGI needs further work on these areas, will work on it after Alpha

Shouldn't we enable **Read Sky Light** by default at least in the meantime? Right now, the SDFGI appearance on outdoor scenes isn't very appealing until you enable that property.

**Comment by FilipLundby** [comment]
> Shouldn't we enable **Read Sky Light** by default at least in the meantime? Right now, the SDFGI appearance on outdoor scenes isn't very appealing until you enable that property.

It would be nice if it was possible to control the shadow color. When **Read sky light** is turned on shadows contains a lot of blue (see image 1). Turning **Read sky light** off makes shadows go black (see image 2).

I am aiming for a more neutral look, like image 3. **I would expect the shadow color of the DirectionalLight to be the shadow color**, but it doesn't seem to do anything. 

The argument might be that the blueish color is more realistic and this might be true, but as a game developer I'm not always going for realism. For winter scene this would be perfect, but for warm summer day it feels out of place. 

Just my two cents, other then that, it looks amazing! :)  

**1. Blue sky and Read sky light on**
image

**2. Read sky light off**
image

**3. Gray sky and Read sky light on**
image

**What the scene and sky looks like**
image

**Comment by Calinou** [comment]
> It would be nice if it was possible to control the shadow color. When Read sky light is turned on shadows contains a lot of blue (see image 1). Turning Read sky light off makes shadows go black (see image 2).

You don't adjust the shadow color. Instead, you adjust the ambient light color. To do this, reduce **Sky Contribution** in the Environment settings (and optionally set a plain color instead of sourcing the sky for ambient light).

**Comment by Calinou** [comment]
> Thank you for the feedback @Calinou. I am probably missing something, because your description sounds exactly as how I set up the environment. Ambient color is set to neutral gray and ambient sky contribution is set to 0.

This turned out to be a (likely) bug: https://github.com/godotengine/godot/issues/54681

**Review by akien-mga — APPROVED** [review]
Approved in PR review meeting.

**Pull request merged**
